### DISCUSSION TITLE
chat-history-memory-management

### DISCUSSION THREAD

**Discussion by mylee04** [microsoft/promptflow#2016] [General]
I got an error message, 'context_length_exceeded,' while attempting to store the entire chat history between the chat model and myself. I understand that upgrading from GPT-3.5 to GPT-4 would allow for handling more than 8k tokens, but I see this as only a temporary solution. I'm seeking a sustainable approach. Do you have any recommendations? 

For instance, perhaps storing only the most recent 10 Q&A messages?
<img width="590" alt="2024-02-13_13-06-37" src="https://github.com/microsoft/promptflow/assets/86863395/499e034d-bfd4-4625-8e85-6402291719ec">

**Comment by brynn-code** [comment]
Hi, sorry for the late response and thanks for reaching us. Since different users can have very different needs for chat history management, prompt flow has no plan to support chat history management in short term, customers are encouraged to use third-party tools to achieve their goal such as langchain chat history management, OpenAI threads or other methods like the DB or something else.
